pkg/test/eventually.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"testing"
+	"time"
+)
+
+// A Condition is a function that returns true when a test condition is satisfied.
```

Comment:
I understand eventually is useful in certain circumstances but I want to add a wary for strictly limiting its use, of no alternative better solution is available. 

we should encourage implementing more efficient and customized waiting cycling using channels instead of this generalized backoff logic. 

Also factor in that go discourage common libraries.

Reply:
No, the whole goal is that we do _not_ encourage customized waiting using channels. That means we wind up with duplicate code that is not necessarily trivial to get correct all over the place. This is _exactly_ the kind of logic that should be abstracted away into a function.